src/io_header.c

```diff
@@ -216,10 +216,52 @@ MQ1_fields allocate_MQ1_fields(unsigned int nheaders)
   }
 
   /*threshold is float[8]*/
-  mq1_fields.threshold = (float *) malloc(
-    sizeof(float) * MQ1_FLOAT_LEN_THRESHOLD * mq1_fields.max_length);
-  if (mq1_fields.threshold == NULL) {
-    perror("Memory allocation error for threshold");
+  mq1_fields.threshold0 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold0 == NULL) {
+    perror("Memory allocation error for threshold0");
+    exit(1);
+  }
+  mq1_fields.threshold1 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold1 == NULL) {
+    perror("Memory allocation error for threshold1");
+    exit(1);
+  }
+  mq1_fields.threshold2 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold2 == NULL) {
+    perror("Memory allocation error for threshold2");
+    exit(1);
+  }
+  mq1_fields.threshold3 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold3 == NULL) {
+    perror("Memory allocation error for threshold3");
+    exit(1);
+  }
+  mq1_fields.threshold4 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold4 == NULL) {
+    perror("Memory allocation error for threshold4");
+    exit(1);
+  }
+  mq1_fields.threshold5 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold5 == NULL) {
+    perror("Memory allocation error for threshold5");
+    exit(1);
+  }
+  mq1_fields.threshold6 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold6 == NULL) {
+    perror("Memory allocation error for threshold6");
+    exit(1);
+  }
+  mq1_fields.threshold7 =
+    (float *) malloc(sizeof(float) * mq1_fields.max_length);
+  if (mq1_fields.threshold7 == NULL) {
+    perror("Memory allocation error for threshold7");
     exit(1);
   }
```

Comment:
A lot of duplicated codes here, could we have a static inline function for these logic?